Mathlib/Lean/Meta/Simp.lean

```diff
@@ -159,8 +159,10 @@ Given a simplifier `S : Expr → MetaM Simp.Result`,
 and an expression `e : Expr`, run `S` on the type of `e`, and then
 convert `e` into that simplified type, using a combination of type hints and `Eq.mp`.
 -/
-def simpType (S : Expr → MetaM Simp.Result) (e : Expr) : MetaM Expr := do
-  match ← S (← inferType e) with
+def simpType (S : Expr → MetaM Simp.Result) (e : Expr) (typeHint : Option Expr := none) :
+    MetaM Expr := do
+  let type ← typeHint.getDM  (inferType e)
```

Comment:
Rather than saying it's a hint, how about saying it's the (optional) type? Using a question mark like this is fairly standard in Lean APIs. Could you also add to the docstring that `type?` should be defeq to the type of `e`?
```suggestion
def simpType (S : Expr → MetaM Simp.Result) (e : Expr) (type? : Option Expr := none) :
    MetaM Expr := do
  let type ← type?.getDM (inferType e)
```